### PR TITLE
feat(program): add openDaysAfterCoEndDate to programs

### PR DESCRIPTION
**Pull request by tomzemp**
This PR adds "Open days after category option end date" field to event and tracker program forms (corresponding to openDaysAfterCoEndDate). This field should be conditionally displayed if a category combination is not none. See https://jira.dhis2.org/browse/DHIS2-12118.

Note: after I added conditional logic rules into field-rules.js, I noticed that they were not applied except on initial render. I therefore reconnected ProgramForm component in EventProgramStepperContent.js to get them to rerender with field updates. If there's some other approach you'd like implemented, let me know.

### CODE REVIEW

File: src/EditModel/event-program/EventProgramStepperContent.js [hunk 1]
```diff
@@ -20,13 +20,20 @@ import {editProgramStageField} from "./tracker-program/program-stages/actions";
 
 const eventProgramFields = fieldOrder.for('eventProgram');
 const eventProgramStageFields = fieldOrder.for('eventProgramStage');
+
+const enhance = compose(
+    mapPropsStream(props$ => props$
+        .combineLatest(eventProgramStore.map(get('program')), (props, model) => ({ ...props, model })),
+    ),
+    createFieldConfigsFor('program', eventProgramFields, undefined, null, true, 'eventProgram'),
```
[the remark this answers is not in the document]
Comment (tomzemp) [reply src/EditModel/event-program/EventProgramStepperContent.js]: fixed

**Comment by tomzemp** [comment]
Thanks @Birkbjo: I've added `categoryCombo[id,name]` into the program metadata and have fixed the incorrect false argument in createFieldConfigsFor line.

I'd be happy to refactor to use createFormFor if you'd prefer. Just let me know / if there's any other changes you'd like.

**Comment by Birkbjo** [comment]
> Thanks @Birkbjo: I've added `categoryCombo[id,name]` into the program metadata and have fixed the incorrect false argument in createFieldConfigsFor line.
> 
> I'd be happy to refactor to use createFormFor if you'd prefer. Just let me know / if there's any other changes you'd like.

Thanks again for implementing this! I think we can keep this as is for now, to keep the surface area as little as possible. Especially since after a quick investigation, it does not look like we can easily use `createFormFor`-for the programStage part of the form, since the `createFormFor` needs a source-observable, and we already have access to the `programStage` as a prop. 

So now it should just be a matter of rebasing on master and fixing conflicts :)

**Comment by benguaraldi** [comment]
@tomzemp I can indeed merge this.

@Birkbjo, do you want to look over @tomzemp's changes before I do?

**Comment by Birkbjo** [comment]
Yeah, lets merge. I've also seen the checks failing on forks - but I believe that is just the PR-checks itself, not once its merged to the repo. But this repo do not have "unsafe"-checks.

**Comment by benguaraldi** [comment]
Thanks, @Birkbjo!

**Pull request merged**